app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -1360,6 +1753,71 @@ class AppointmentRepositoryAndroidTest {
     assertThat(overdueAppointments).containsExactlyElementsIn(expectedAppointments)
   }
 
+  @Test
+  fun patients_without_blood_sugars_should_not_be_shown_when_fetching_overdue_appointments() {
+
+    val currentDate = LocalDate.parse("2018-01-05")
+
+    fun createAppointmentRecord(
+        patientUuid: UUID,
+        bloodSugarUuid: UUID?,
+        appointmentUuid: UUID
+    ): RecordAppointment {
+      val patientProfile = testData.patientProfile(patientUuid = patientUuid, generatePhoneNumber = true)
+
+      val bloodSugarMeasurement = if (bloodSugarUuid != null) {
+        testData.bloodSugarMeasurement(
+            uuid = bloodSugarUuid,
+            patientUuid = patientUuid,
+            facilityUuid = facility.uuid,
+            userUuid = userUuid,
+            reading = BloodSugarReading("256", Random),
+            recordedAt = Instant.parse("2018-01-01T00:00:00Z"),
+            deletedAt = null
+        )
+      } else null
+
+      val appointment = testData.appointment(
+          uuid = appointmentUuid,
+          patientUuid = patientUuid,
+          facilityUuid = facility.uuid,
+          scheduledDate = LocalDate.parse("2018-01-04"),
+          status = Scheduled,
+          cancelReason = null,
+          remindOn = null,
+          agreedToVisit = null
+      )
+
+      return RecordAppointment(patientProfile, null, bloodSugarMeasurement, appointment)
+    }
+
+    // given
+    val withBloodSugar = createAppointmentRecord(
+        patientUuid = UUID.fromString("417c19d3-68a0-4936-bc4f-5b7c2a73ccc7"),
+        bloodSugarUuid = UUID.fromString("3414fd9a-8b30-4850-9f8f-3de9305dcb6c"),
+        appointmentUuid = UUID.fromString("053f2f73-b693-420c-a9c6-d8aae1c77395")
+    )
+
+    val withoutBloodSugar = createAppointmentRecord(
+        patientUuid = UUID.fromString("0af5c909-551b-448d-988e-b00b3304f738"),
+        bloodSugarUuid = null,
+        appointmentUuid = UUID.fromString("e58cfd76-aaeb-42a8-8bf1-4c71614c6288")
+    )
```

Comment:
This is not getting inserted into the database since we are saving `listOf(withBloodSugar, withBloodSugar)`.